File: packages/host4_flutter_aivoice/ios/FloatingWindowPlugin_副本/XHFloatWindow.m
//
//  XHFloatWindow.m
//  XHFloatingWindow
//
//  Created by Xinhou Jiang on 14/1/17.
//  Copyright © 2017年 Xinhou Jiang. All rights reserved.
//

#import "XHFloatWindow.h"
#import "XHFloatWindowSingleton.h"
#import "XHFloatWindowController.h"

@interface XHFloatWindow ()

@end

@implementation XHFloatWindow

+(BOOL)xh_isShowing{
    return [[XHFloatWindowSingleton Ins] isShowing];
}

+ (void)xh_addWindowOnTarget:(id)target onClick:(void (^)())callback {
    [[XHFloatWindowSingleton Ins] xh_addWindowOnTarget:target onClick:callback];
}

+ (void)xh_setWindowSize:(float)size {
    [[XHFloatWindowSingleton Ins] xh_setWindowSize:size];
}

+ (void)xh_setHideWindow:(BOOL)hide {
    [[XHFloatWindowSingleton Ins] xh_setHideWindow:hide];
}

+ (void)xh_setBackgroundImage:(NSString *)imageName forState:(UIControlState)UIControlState {
    [[XHFloatWindowSingleton Ins] xh_setBackgroundImage:imageName forState:UIControlState];
}

+ (void)xh_setGifImageName:(NSString *)gifName playOnClick:(BOOL)playOnClick {
    [[XHFloatWindowSingleton Ins] xh_setGifImageName:gifName playOnClick:playOnClick];
}

+ (CGRect)xh_getCurrentFrame {
    return [[XHFloatWindowSingleton Ins] xh_getCurrentFrame];
}

+ (void)xh_setInitialPosition:(CGPoint)position {
    [[XHFloatWindowSingleton Ins] xh_setInitialPosition:position];
}

+ (void)xh_destroyWindow {
    [[XHFloatWindowSingleton Ins] xh_destroyWindow];
}

+ (void)xh_rebuildWindowOnTarget:(id)target onClick:(void (^)())callback {
    [[XHFloatWindowSingleton Ins] xh_rebuildWindowOnTarget:target onClick:callback];
}

+ (BOOL)xh_isShrunk {
    return [[XHFloatWindowSingleton Ins] xh_isShrunk];
}

+ (void)xh_setShrunk:(BOOL)isShrunk {
    [[XHFloatWindowSingleton Ins] xh_setShrunk:isShrunk];
}

+ (BOOL)xh_getSavedShrunkState {
    // 如果窗口正在显示，直接返回当前状态
    if ([XHFloatWindowSingleton Ins].floatVC) {
        return [XHFloatWindowSingleton Ins].floatVC.isShrunk;
    }
    
    // 如果窗口已销毁，返回缓存的状态
    return [XHFloatWindowSingleton Ins].savedShrunkState;
}
@end

File: packages/host4_flutter_aivoice/ios/Classes/FloatingWindow/XHFloatWindow.swift
import UIKit

/// 悬浮窗接口（静态方法转发到单例）
class XHFloatWindow: NSObject {

  @discardableResult
  static func xh_isShowing() -> Bool {
    return XHFloatWindowSingleton.shared.isShowing
  }

  static func xh_addWindowOnTarget(_ target: Any, onClick callback: (() -> Void)?) {
    XHFloatWindowSingleton.shared.xh_addWindowOnTarget(target, onClick: callback)
  }

  static func xh_setWindowSize(_ size: CGFloat) {
    XHFloatWindowSingleton.shared.xh_setWindowSize(size)
  }

  static func xh_setHideWindow(_ hide: Bool) {
    XHFloatWindowSingleton.shared.xh_setHideWindow(hide)
  }

  static func xh_setBackgroundImage(_ imageName: String?, for state: UIControl.State) {
    XHFloatWindowSingleton.shared.xh_setBackgroundImage(imageName, for: state)
  }

  static func xh_setInitialPosition(_ position: CGPoint) {
    XHFloatWindowSingleton.shared.xh_setInitialPosition(position)
  }

  static func xh_getCurrentFrame() -> CGRect {
    return XHFloatWindowSingleton.shared.xh_getCurrentFrame()
  }

  static func xh_destroyWindow() {
    XHFloatWindowSingleton.shared.xh_destroyWindow()
  }

  static func xh_rebuildWindowOnTarget(_ target: Any, onClick callback: (() -> Void)?) {
    XHFloatWindowSingleton.shared.xh_rebuildWindowOnTarget(target, onClick: callback)
  }

  static func xh_isShrunk() -> Bool {
    return XHFloatWindowSingleton.shared.xh_isShrunk()
  }

  static func xh_setShrunk(_ shrunk: Bool) {
    XHFloatWindowSingleton.shared.xh_setShrunk(shrunk)
  }

  static func xh_getSavedShrunkState() -> Bool {
    return XHFloatWindowSingleton.shared.savedShrunkState
  }
}
